src/lib/uri/uri.h

```diff
@@ -51,6 +101,20 @@ uri_destroy(struct uri *uri);
 int
 uri_format(char *str, int len, const struct uri *uri, bool write_password);
 
+/**
+ * Return @a uri option value by given @a idx. If option with @a name is not
+ * exist or @a idx less than zero or greater than or equal to URI option
+ * values count return NULL.
+ */
+const char *
+uri_option_value_by_idx(const struct uri *uri, const char *name, int idx);
```

Comment:
Let's name it simply `uri_query_param` or `uri_param` (without `_value_by_idx` suffix), depending on how you'll call the corresponding struct (it's okay to give a function the same name as a struct in C).